src/index.js

```diff
@@ -8,3 +10,32 @@ export {
 
 export * from './service';
 export * from './message';
+
+export default (config) => {
+  const internalLogger = new Logger(config);
+
+  return (label) => {
+    const labelledLogger = internalLogger.Label(label);
+
+    const logger = level => (...args) => {
+      let msg;
+
+      if (args.length === 2) {
```

Comment:
Yeah, this is more like a demo version and meant not to be the final. Since I'm not sure whether there is going to be other message, I just check the arguments length here.

But I'm not sure we should always construct a new message like this. Our major API should cover 90% of the usage while leaving ability to customize and extend. My first thought on this is to accept two types of argument in the logging function.

```js
// this will create `LogMessage` internally
logger.warn('something', { additionalInfo });

// function detect argument is a instance of `LogMessage`, skip the default behavior
logger.warn(new PaymentErrorMessage('something', { additionalInfo }));
```

But in this approach, we still need to distinguish `LogMessage` and `ErrorMessage` properly.